src/app/users/users.service.ts

```diff
@@ -190,9 +193,30 @@ export class UsersService {
         // Make copy of user so UI doesn't change until DB change succeeds
         const tempUser = { ...user, roles: newRoles };
         observers.push(this.couchService.put('_users/org.couchdb.user:' + tempUser.name, tempUser));
+        this.sendNotifications(user);
       }
       return observers;
     }, [])).pipe(map((responses) => this.stateService.requestData(this.dbName, 'local')));
   }
 
+  sendNotifications(user) {
+    const link = '/manager/users';
+    const notificationDoc = {
+      user: user._id,
+      'message': 'You were assigned a new role',
+      link,
+      linkParams: { activeTab: 'taskTab' },
+      'type': 'newRole',
+      'priority': 1,
+      'status': 'unread',
+      'time': this.couchService.datePlaceholder,
+      userPlanetCode: user.userPlanetCode
+    };
+    return this.couchService.findAll(
+      'notifications',
+    ).pipe(
+      switchMap((res: any[]) => res.length > 0 ? this.couchService.updateDocument('notifications', notificationDoc) : of({}))
+    ).subscribe((users: any) => users);
```

Comment:
Ideally this would not include the `.subscribe` and `sendNotifications` would be called as part of a `switchMap` after the role change was completed on the database.